client/src/pages/Friends.js

```diff
@@ -1,27 +1,99 @@
-import React from "react";
-import { Box, Typography } from "@material-ui/core";
+import React, { useState } from "react";
+import { Box, Typography, Tabs,Tab } from "@material-ui/core";
 import { makeStyles } from "@material-ui/core/styles";
 
+import Followers from '../components/Followers'
+import Following from "components/Following";
+import rose from '../img/roseBarron.png';
+import david from '../img/davidRichardson.png'
+import christine from '../img/christineSmith.png'
+import jennifer from '../img/jenniferFordham.png'
+import rebbeca from  '../img/rebbecaSchindler.png'
+
 const useStyles = makeStyles((theme) => ({
-  friendsContainer: {
-    width: "100%",
-    margin: "10rem 0",
-    display: "flex",
-    justifyContent: "center",
+  h1:{
+    textAlign:'center',
+    fontSize:'1.5rem',
+    marginTop:theme.spacing(10),
+    marginBottom:theme.spacing(6),
+    fontWeight:'bold'
+  },
+  tabs: {
+    maxWidth:'800px',
+    width:'60%',
+    margin:'0 auto',
   },
-  textPlaceholder: {
-    fontSize: "2rem",
+  tab:{
+    width:'50%',
+    maxWidth:'none'
   },
+  tabsContainer:{
+    maxWidth:'800px',
+    width:'60%',
+    margin:'0 auto',
+    backgroundColor:'white'
+  }
+  
 }));
 
-const Friends = () => {
+const Friends = (props) => {
   const classes = useStyles();
-
+  const [selectedPage, setSelectedPage] = useState(0);
+  const [followersList, setFollowers]=useState(
+    [
+    {name:'David Richardson', img:david,},
+    {name:'Christine Smith', img:christine,},
+    {name:'Jennifer Fordham', img:jennifer,},
+    {name:'Rebbeca Schindler', img:rebbeca,}
+    ])
+  const [followingsList, setFollowings]=useState(
+    [
+    {name:'Rose Barron', img:rose},
+    ])
+  const handleTabChange = (e, newValue) => {
+    setSelectedPage(newValue);
+  }
+  const handleFollow=(person)=>{
+    const newFollowingsList=[...followingsList]
+    newFollowingsList.push(person)
+    setFollowings(newFollowingsList)
+  }
+  const handleUnfollow=(person)=>{
+    const newFollowingsList=[...followingsList]
+    newFollowingsList.pop(person)
+    setFollowings(newFollowingsList)
+  }
```

Comment:
So a toggle function would check if the followingsList contains the person. If so they would remove them from the array, if not they would add that to the array (also change newFollowingsList to let as you are mutating it) :

```
  const toggleFollow = (person) => {
    let newFollowingsList = [...followingsList];
    if (followingsList.includes(person)) newFollowingsList.pop(person);
    else newFollowingsList.push(person);
    setFollowings(newFollowingsList);
  };
```

Reply:
So pop doesn't actually work like this  in JavaScript, it just removes the last item from the array.

Given that we probably need to be making API calls to add or remove followers, I think in this case it might actually be better to keep it as separate functions. You could also then also pass the index of the person to be removed, which would make removing from the array more efficient.

Reply:
That's true, the function would get too large if API calls were included as well.

What if we created a simple toggle function like this?
```
  const toggleFollow = (person) => {
      followingsList.includes(person) ? handleUnfollow(person) : handleFollow(person);
  }
```

That way we could separate the concerns of the follow vs unfollow logic, but still have the benefit of only having to pass one function to `<Followers />` & `<Following />`.

Reply:
yeah, I like that!